src/applications/travel-pay/tests/containers/TravelPayStatusApp.unit.spec.jsx

```diff
@@ -60,17 +76,126 @@ describe('App', () => {
+      initialState: getData(),
+      path: `/`,
+      reducers: reducer,
+    });
+    expect(
+      await screenFeatureToggle.getByTestId('travel-pay-loading-indicator'),
+    ).to.exist;
+  });
+
+  it('renders a login prompt for an unauthenticated user', async () => {
+    const screen = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: false,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+    expect(await screen.findByText('Log in to view your travel claims')).to
+      .exist;
+  });
+
+  it('shows the login modal when clicking the login prompt', async () => {
+    const { container } = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: true,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    expect($('va-loading-indicator', container)).to.exist;
+  });
+
+  it('handles a failed fetch of claims', async () => {
+    global.fetch.restore();
+    mockApiRequest({ error: 'oh no' }, false);
+
+    const screen = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: true,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    await waitFor(() => {
+      expect(screen.queryAllByTestId('travel-claim-details').length).to.eq(0);
+    });
+  });
+
+  it('shows the login modal when clicking the login prompt', async () => {
+    const { container } = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: false,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    fireEvent.click($('va-button', container));
+    // TODO: make this check for the modal itself
+    expect($('va-button', container)).to.exist;
+  });
+
+  it('sorts the claims correctly using the select-option', async () => {
+    const screen = renderWithStoreAndRouter(<App />, {
+      initialState: getData({
+        areFeatureTogglesLoading: false,
+        hasFeatureFlag: true,
+        isLoggedIn: true,
+      }),
+      path: `/`,
+      reducers: reducer,
+    });
+
+    userEvent.selectOptions(screen.getByRole('combobox'), ['oldest']);
+    expect(screen.getByRole('option', { name: 'Oldest' }).selected).to.be.true;
+    fireEvent.click(document.querySelector('va-button[text="Sort"]'));
+
+    await waitFor(() => {
+      expect(screen.getAllByTestId('travel-claim-details').length).to.eq(2);
+      // TODO: figure out why this isn't re-rendering the list properly.
+      // expect(
+      //   screen.getAllByTestId('travel-claim-details')[0].textContent,
+      // ).to.eq('Thursday, February 22, 2024 at 11:45 AM appointment');
```

Comment:
I'm not sure if this is due to `va-button` being the new web component version or something else, but it seems like the click event above this isn't properly re-rendering the list with the proper order

Reply:
Ticket created: department-of-veterans-affairs/va.gov-team#83449

Reply:
Pretty sure I fixed the test [here](https://github.com/department-of-veterans-affairs/vets-website/pull/29817/commits/cd674e89e294b9c3b15d92b026ffc670ffdeea45). We'll see how it looks in dev/staging once integration woes are fixed